hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieFileGroupReaderBasedParquetFileFormat.scala

```diff
@@ -129,20 +138,15 @@ class HoodieFileGroupReaderBasedParquetFileFormat(tableState: HoodieTableState,
       file.partitionValues match {
         // Snapshot or incremental queries.
         case fileSliceMapping: HoodiePartitionFileSliceMapping =>
-          val filePath = sparkAdapter.getSparkPartitionedFileUtils.getPathFromPartitionedFile(file)
-          val filegroupName = if (FSUtils.isLogFile(filePath)) {
-            FSUtils.getFileId(filePath.getName).substring(1)
-          } else {
-            FSUtils.getFileId(filePath.getName)
-          }
+          val filegroupName = FSUtils.getFileIdFromFilePath(sparkAdapter
+            .getSparkPartitionedFileUtils.getPathFromPartitionedFile(file))
           fileSliceMapping.getSlice(filegroupName) match {
             case Some(fileSlice) if !isCount =>
               if (requiredSchema.isEmpty && !fileSlice.getLogFiles.findAny().isPresent) {
                 val hoodieBaseFile = fileSlice.getBaseFile.get()
                 baseFileReader(createPartitionedFile(fileSliceMapping.getPartitionValues, hoodieBaseFile.getHadoopPath, 0, hoodieBaseFile.getFileLen))
               } else {
-                val readerContext: HoodieReaderContext[InternalRow] = new SparkFileFormatInternalRowReaderContext(
-                  readerMaps)
+                val readerContext = new SparkFileFormatInternalRowReaderContext(parquetFileReader.value, tableState.recordKeyField, filters)
```

Comment:
I remember the original intention is to remove any arguments of the reader context (the reader instance is instantiated inside the `context.getFileRecordIterator`) and there should be no Hudi related parameters.  Any parameters can be passed by the file group reader as an argument to the context function call.

Reply:
The intention is to not have duplicate logic for creating the readers and then passing in a map of already created readers. Now, we create the readers on the executor. 

Before we called buildReaderWithPartitionValues() a few times and had a map from schema hash to PartitionedFile => Iterator[InternalRow]

Now we have a reader that we call read on the executor and can pass in the schema and filters that we want with the file. 

We have removed the limitation that schema and filters be known in the driver

Reply:
We have also not moved any logic for filters to be in the fg reader. That is still handled just by the spark context for now.

tableState.recordKeyField is used only for filter purposes.

We cannot get rid of the parquetFileReader param. That contains only spark specific configs